pkg/apiserver/event/event_test.go

```diff
@@ -0,0 +1,335 @@
+			conf:    nil,
+			kubeCli: nil,
+			archive: nil,
+			event:   mockes,
+		}
+		router = gin.Default()
+		r := router.Group("/api")
+		endpoint := r.Group("/events")
+		endpoint.GET("", s.listEvents)
+		endpoint.GET("/dry", s.listDryEvents)
+		endpoint.GET("/get", s.getEvent)
+	})
+
+	AfterEach(func() {
+		// Add any setup steps that needs to be executed before each test
+	})
+
+	Context("ListEvents", func() {
+		It("empty podNamespace", func() {
+			rr := httptest.NewRecorder()
+			request, err := http.NewRequest(http.MethodGet, "/api/events?podName=testpodNamespace", nil)
+			Expect(err).ShouldNot(HaveOccurred())
+			router.ServeHTTP(rr, request)
+			Expect(rr.Code).Should(Equal(http.StatusInternalServerError))
+		})
+
+		It("success", func() {
+			response := []*core.Event{
+				&core.Event{
+					ID:           0,
+					CreatedAt:    time.Time{},
+					UpdatedAt:    time.Time{},
+					DeletedAt:    nil,
+					Experiment:   "testExperiment",
+					Namespace:    "testNamespace",
+					Kind:         "testKind",
+					Message:      "test",
+					StartTime:    nil,
+					FinishTime:   nil,
+					Duration:     "1m",
+					Pods:         nil,
+					ExperimentID: "testUID",
+				},
+			}
+			rr := httptest.NewRecorder()
+			request, err := http.NewRequest(http.MethodGet, "/api/events?uid=testUID", nil)
+			Expect(err).ShouldNot(HaveOccurred())
+			router.ServeHTTP(rr, request)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(rr.Code).Should(Equal(http.StatusOK))
+			responseBody, err := json.Marshal(response)
+			Expect(rr.Body.Bytes()).Should(Equal(responseBody))
+		})
+
+		It("test err", func() {
+			rr := httptest.NewRecorder()
+			request, err := http.NewRequest(http.MethodGet, "/api/events?uid=err", nil)
+			Expect(err).ShouldNot(HaveOccurred())
+			router.ServeHTTP(rr, request)
+			Expect(rr.Code).Should(Equal(http.StatusInternalServerError))
+		})
+	})
+
+	Context("ListDryEvents", func() {
+		It("success", func() {
+			response := []*core.Event{
+				&core.Event{
+					ID:           0,
+					CreatedAt:    time.Time{},
+					UpdatedAt:    time.Time{},
+					DeletedAt:    nil,
+					Experiment:   "testExperiment",
+					Namespace:    "testNamespace",
+					Kind:         "testKind",
+					Message:      "test",
+					StartTime:    nil,
+					FinishTime:   nil,
+					Duration:     "1m",
+					Pods:         nil,
+					ExperimentID: "testUID",
+				},
+			}
+			rr := httptest.NewRecorder()
+			request, err := http.NewRequest(http.MethodGet, "/api/events/dry?kind=testKind", nil)
+			Expect(err).ShouldNot(HaveOccurred())
+			router.ServeHTTP(rr, request)
+			Expect(rr.Code).Should(Equal(http.StatusOK))
+			responseBody, err := json.Marshal(response)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(rr.Body.Bytes()).Should(Equal(responseBody))
+		})
+
+		It("test err", func() {
+			rr := httptest.NewRecorder()
+			request, err := http.NewRequest(http.MethodGet, "/api/events/dry?kind=err", nil)
+			Expect(err).ShouldNot(HaveOccurred())
```

Comment:
Shoud be `Expect(err).Should(HaveOccurred())`?

Reply:
This is the err returned by "NewRequest" , which should be nil.

Reply:
🤔 Know about it. But it seems that all tested requests returned `nil`. Is it necessary to make this step of assertion?

Reply:
OK, I'll take these out.